airsonic-main/src/main/java/org/airsonic/player/controller/SubsonicRESTController.java

```diff
@@ -1723,7 +1725,9 @@ public void getBookmarks(HttpServletRequest request, HttpServletResponse respons
             b.setChanged(jaxbWriter.convertDate(bookmark.getChanged()));
 
             MediaFile mediaFile = mediaFileService.getMediaFile(bookmark.getMediaFileId());
-            b.setEntry(createJaxbChild(player, mediaFile, username));
+            if (mediaFile != null) {
+                b.setEntry(createJaxbChild(player, mediaFile, username));
```

Comment:
I assume that `entry` is going to be null in this case, wouldn't it be better to skip it entirely with `continue`?

Reply:
I was of two minds here, thinking that empty bookmarks might give a warning of something being amiss in the database without crashing the system. Then again, people probably don't want to see empty bookmarks... therefore I moved the test up to avoid generating empty entries.